src/modules/cli/lib/execute-command.ts

```diff
@@ -0,0 +1,65 @@
+import { fork } from 'node:child_process';
+import EventEmitter from 'node:events';
+import path from 'node:path';
+import { getResourcesPath } from 'src/storage/paths';
+
+function* parseOutput( data: Buffer ): Generator< unknown, void, unknown > {
+	const output = data.toString( 'utf8' );
+
+	for ( const line of output.split( '\n' ) ) {
+		if ( ! line.trim() ) {
+			continue;
+		}
+
+		try {
+			yield JSON.parse( line.trim() );
+		} catch ( error ) {
+			console.log( 'Parsing error', error, line );
+		}
+	}
+}
+
+export function executeCliCommand( args: string[] ) {
+	const cliPath =
+		process.env.NODE_ENV === 'development'
+			? path.join( getResourcesPath(), 'dist', 'cli', 'main.js' )
+			: path.join( getResourcesPath(), 'cli', 'main.js' );
+	// Using Electron's utilityProcess.fork API gave us issues with the child process never exiting
+	const child = fork( cliPath, [ ...args, '--output-format', 'json' ], {
+		stdio: 'pipe',
+	} );
+	const eventEmitter = new EventEmitter();
+
+	child.stdout?.on( 'data', ( data: Buffer ) => {
+		for ( const parsed of parseOutput( data ) ) {
+			eventEmitter.emit( 'data', parsed );
+		}
+	} );
+
+	child.stderr?.on( 'data', ( data: Buffer ) => {
+		for ( const parsed of parseOutput( data ) ) {
+			eventEmitter.emit( 'error', parsed );
+		}
+	} );
+
+	child.on( 'error', ( error ) => {
+		console.error( 'Child process error:', error );
+		eventEmitter.emit( 'error', error );
+	} );
+
+	child.on( 'exit', ( code: number | null ) => {
+		console.log( 'Preview site creation completed with code', code );
```

Comment:
While reviewing https://github.com/Automattic/studio/pull/1223 I noticed that we will have a specific execute command for previews. It will likely make sense to remove this